File: Solution/Medium/0165 Compare Version Numbers.java
class Solution {
    public int compareVersion(String version1, String version2) {
        String[] str1 = version1.split("\\.");
        String[] str2 = version2.split("\\.");

        int len1 = str1.length;
        int len2 = str2.length;

        if (len1 == len2) {
            for (int i = 0; i < len1; i++) {
                if (Integer.parseInt(str1[i]) < Integer.parseInt(str2[i])) {
                    return -1;
                } else if (Integer.parseInt(str1[i]) > Integer.parseInt(str2[i])) {
                    return 1;
                }
            }
            return 0;
        } else if (len1 < len2) {
            int i = 0;
            for (; i < len1; i++) {
                if (Integer.parseInt(str1[i]) < Integer.parseInt(str2[i])) {
                    return -1;
                } else if (Integer.parseInt(str1[i]) > Integer.parseInt(str2[i])) {
                    return 1;
                }
            }
            for (int j = i; j < len2; j++) {
                if (Integer.parseInt(str2[j]) > 0) {
                    return -1;
                }
            }
            return 0;
        } else {
            int i = 0;
            for (; i < len2; i++) {
                if (Integer.parseInt(str1[i]) < Integer.parseInt(str2[i])) {
                    return -1;
                } else if (Integer.parseInt(str1[i]) > Integer.parseInt(str2[i])) {
                    return 1;
                }
            }
            for (int j = i; j < len1; j++) {
                if (Integer.parseInt(str1[j]) > 0) {
                    return 1;
                }
            }
            return 0;
        }
    }
}

File: Solution/Medium/0165 Compare Version Numbers.kt
class Solution {
    fun compareVersion(version1: String, version2: String): Int {
        val v1 = version1.split(".").map { it.toInt() }.toMutableList()
        val v2 = version2.split(".").map { it.toInt() }.toMutableList()
        while (v1.size != v2.size) {
            if (v1.size < v2.size) v1.add(0)
            else v2.add(0)
        }
        v1.zip(v2).forEach { (val1, val2) ->
            if (val1 > val2) return 1
            else if (val1 < val2) return -1
        }
        return 0
    }
}
